Can't use helm provider v1 with plan & apply on different machines

Hello

There is an issue with running plan & apply on different build agents in a CI CD pipeline. It is document [here](https://github.com/terraform-providers/terraform-provider-helm/issues/319#issuecomment-523766938)

The use of the **home** key in the provider worked fine in helm provider < 1.0. for example:

```hcl
provider "helm" {
  debug           = true
  version         = "~> 0.10"
  namespace       = "kube-system"
  service_account = kubernetes_service_account.tiller_sa.metadata.0.name
  home            = "${abspath(path.root)}/.helm"

  kubernetes {
    host                   = azurerm_kubernetes_cluster.aks.kube_config.0.host
    client_certificate     = base64decode(azurerm_kubernetes_cluster.aks.kube_config.0.client_certificate)
    client_key             = base64decode(azurerm_kubernetes_cluster.aks.kube_config.0.client_key)
    cluster_ca_certificate = base64decode(azurerm_kubernetes_cluster.aks.kube_config.0.cluster_ca_certificate)
    load_config_file       = false
  }
}
```

However, when upgrading the provider to version 1.0, the **home** key is invalid anymore
```console
terraform plan
Acquiring state lock. This may take a few moments...

Error: Unsupported argument

  on init.tf line 40, in provider "helm":
  40:   home            = "${abspath(path.root)}/.helm"

An argument named "home" is not expected here.
```

This key can be removed and terraform will work fine **as long as the plan & apply both done on the same machine (e.g. developer's laptop)**
But when running the plan and apply on different machines (for example, during a CI / CD process) this will break since the **/.helm** folder doesn't exist on the agent, causing the deployment of the helm release to fail. for example:

```console
# plan phase

2020-02-20T11:55:00.2426615Z [1m  # helm_release.phippyandfriends["parrot"][0m will be created[0m[0m
2020-02-20T11:55:00.2427141Z [0m  [32m+[0m[0m resource "helm_release" "phippyandfriends" {
2020-02-20T11:55:00.2427646Z       [32m+[0m [0m[1m[0matomic[0m[0m                = false
2020-02-20T11:55:00.2428208Z       [32m+[0m [0m[1m[0mchart[0m[0m                 = "parrot"
2020-02-20T11:55:00.2428911Z       [32m+[0m [0m[1m[0mcleanup_on_fail[0m[0m       = false
2020-02-20T11:55:00.2429613Z       [32m+[0m [0m[1m[0mdependency_update[0m[0m     = false
2020-02-20T11:55:00.2430106Z       [32m+[0m [0m[1m[0mdisable_crd_hooks[0m[0m     = false
2020-02-20T11:55:00.2430586Z       [32m+[0m [0m[1m[0mdisable_webhooks[0m[0m      = false
2020-02-20T11:55:00.2431122Z       [32m+[0m [0m[1m[0mforce_update[0m[0m          = false
2020-02-20T11:55:00.2432099Z       [32m+[0m [0m[1m[0mid[0m[0m                    = (known after apply)
2020-02-20T11:55:00.2432645Z       [32m+[0m [0m[1m[0mmax_history[0m[0m           = 0
2020-02-20T11:55:00.2436642Z       [32m+[0m [0m[1m[0mmetadata[0m[0m              = (known after apply)
2020-02-20T11:55:00.2437794Z       [32m+[0m [0m[1m[0mname[0m[0m                  = "parrot"
2020-02-20T11:55:00.2440461Z       [32m+[0m [0m[1m[0mnamespace[0m[0m             = "phippyandfriends"
2020-02-20T11:55:00.2449192Z       [32m+[0m [0m[1m[0mrecreate_pods[0m[0m         = false
2020-02-20T11:55:00.2450215Z       [32m+[0m [0m[1m[0mrender_subchart_notes[0m[0m = true
2020-02-20T11:55:00.2450894Z       [32m+[0m [0m[1m[0mreplace[0m[0m               = false
2020-02-20T11:55:00.2451419Z       [32m+[0m [0m[1m[0mrepository[0m[0m            = "***"
2020-02-20T11:55:00.2451948Z       [32m+[0m [0m[1m[0mreset_values[0m[0m          = false
2020-02-20T11:55:00.2452482Z       [32m+[0m [0m[1m[0mreuse_values[0m[0m          = false
2020-02-20T11:55:00.2453060Z       [32m+[0m [0m[1m[0mskip_crds[0m[0m             = false
2020-02-20T11:55:00.2453776Z       [32m+[0m [0m[1m[0mstatus[0m[0m                = "deployed"
2020-02-20T11:55:00.2454491Z       [32m+[0m [0m[1m[0mtimeout[0m[0m               = 300
2020-02-20T11:55:00.2454986Z       [32m+[0m [0m[1m[0mverify[0m[0m                = false
2020-02-20T11:55:00.2455494Z       [32m+[0m [0m[1m[0mversion[0m[0m               = "v0.5.0"
2020-02-20T11:55:00.2456021Z       [32m+[0m [0m[1m[0mwait[0m[0m                  = true
2020-02-20T11:55:00.2456689Z 
2020-02-20T11:55:00.2468851Z       [32m+[0m [0mset {
2020-02-20T11:55:00.2469468Z           [32m+[0m [0m[1m[0mname[0m[0m  = "image.repository"
2020-02-20T11:55:00.2470036Z           [32m+[0m [0m[1m[0mvalue[0m[0m = "***.azurecr.io/parrot"
2020-02-20T11:55:00.2470298Z         }
2020-02-20T11:55:00.2470677Z       [32m+[0m [0mset {
2020-02-20T11:55:00.2471118Z           [32m+[0m [0m[1m[0mname[0m[0m  = "ingress.alias"
2020-02-20T11:55:00.2472657Z           [32m+[0m [0m[1m[0mvalue[0m[0m = "phippyandfriends.dvps.***.guru"
2020-02-20T11:55:00.2473543Z         }
2020-02-20T11:55:00.2474072Z       [32m+[0m [0mset {
2020-02-20T11:55:00.2474823Z           [32m+[0m [0m[1m[0mname[0m[0m  = "ingress.basedomain"
2020-02-20T11:55:00.2475892Z           [32m+[0m [0m[1m[0mvalue[0m[0m = (known after apply)
2020-02-20T11:55:00.2476369Z         }
2020-02-20T11:55:00.2476593Z     }
...
2020-02-20T11:55:00.2559391Z [0m[1mPlan:[0m 13 to add, 0 to change, 0 to destroy.[0m
2020-02-20T11:55:00.2559598Z 
2020-02-20T11:55:00.2559986Z ------------------------------------------------------------------------
2020-02-20T11:55:00.2560232Z 
2020-02-20T11:55:00.2560652Z This plan was saved to: 503-dvps.plan
2020-02-20T11:55:00.2560853Z 
2020-02-20T11:55:00.2561070Z To perform exactly these actions, run the following command to apply:
2020-02-20T11:55:00.2561531Z     terraform apply "503-dvps.plan"
2020-02-20T11:55:00.2561765Z 
2020-02-20T11:55:00.3165350Z ##[section]Finishing: Terraform Dry Run (Plan)
```

failure in apply phase:
```console
# apply phase

2020-02-20T12:21:11.9461719Z [1m[31mError: [0m[0m[1mrepo *** not found[0m
2020-02-20T12:21:11.9461981Z 
2020-02-20T12:21:11.9462420Z [0m  on main.tf line 151, in resource "helm_release" "phippyandfriends":
2020-02-20T12:21:11.9462846Z  151: resource "helm_release" "phippyandfriends" [4m{[0m
2020-02-20T12:21:11.9463198Z [0m
2020-02-20T12:21:11.9463508Z [0m[0m
2020-02-20T12:21:11.9473599Z [31m
2020-02-20T12:21:11.9474198Z [1m[31mError: [0m[0m[1mrepo *** not found[0m
2020-02-20T12:21:11.9474291Z 
2020-02-20T12:21:11.9476491Z [0m  on main.tf line 151, in resource "helm_release" "phippyandfriends":
2020-02-20T12:21:11.9477223Z  151: resource "helm_release" "phippyandfriends" [4m{[0m
2020-02-20T12:21:11.9477766Z [0m
2020-02-20T12:21:11.9478336Z [0m[0m
2020-02-20T12:21:11.9488647Z [31m
2020-02-20T12:21:11.9489190Z [1m[31mError: [0m[0m[1mrepo *** not found[0m
2020-02-20T12:21:11.9489271Z 
2020-02-20T12:21:11.9489671Z [0m  on main.tf line 151, in resource "helm_release" "phippyandfriends":
2020-02-20T12:21:11.9490010Z  151: resource "helm_release" "phippyandfriends" [4m{[0m
2020-02-20T12:21:11.9490246Z [0m
2020-02-20T12:21:11.9490588Z [0m[0m
2020-02-20T12:21:11.9500783Z [31m
2020-02-20T12:21:11.9505476Z [1m[31mError: [0m[0m[1mrepo *** not found[0m
2020-02-20T12:21:11.9505598Z 
2020-02-20T12:21:11.9505922Z [0m  on main.tf line 151, in resource "helm_release" "phippyandfriends":
2020-02-20T12:21:11.9506204Z  151: resource "helm_release" "phippyandfriends" [4m{[0m
2020-02-20T12:21:11.9506420Z [0m
2020-02-20T12:21:11.9506611Z [0m[0m
2020-02-20T12:21:12.0553033Z ##[error]Bash exited with code '1'.
2020-02-20T12:21:12.0564378Z ##[section]Finishing: Deploy (Terraform Apply)
```

**please note that it is not a problem with the configuration / etc**:
- it is working on a dev machine, where both plan and apply running on the same machine
- it is working with helm provider < 1.0, meaning there is no problem in the actual terraform code
- reverting back to the old version fix the problem without any other changes

How can I accomplish the same scenario (plan & apply on separate machines) using the new provider version (and finally remove tiller 😄 )?

Omer

--- a/helm/data_repository.go
+++ b/helm/data_repository.go
@@ -11,6 +11,8 @@ import (
 
 func dataRepository() *schema.Resource {
 	return &schema.Resource{
+		DeprecationMessage: "This resource is deprecated and will be removed in the next major version.",
+
 		Read: dataRepositoryRead,
 		Schema: map[string]*schema.Schema{
 			"name": {

--- a/website/docs/d/repository.html.markdown
+++ b/website/docs/d/repository.html.markdown
@@ -8,6 +8,8 @@ description: |-
 
 # Data Source: helm_repository
 
+!> **Warning:** This resource is deprecated and will be removed in the next major version.
+
 A chart repository is a location where packaged charts can be stored and shared.
 
 `helm_repository` describes a helm repository.
@@ -50,7 +52,3 @@ The `metadata` block supports:
 
 * `name` - Name of the repository read from the home.
 * `url` - URL of the repository read from the home.
-
-## Old resource helm_repository
-
-Before 0.9.0 `helm_repository` was a resource and not a data source. The old resource is now a shim to the data source to preserve backwards compatibility. As the use of the resource is deprecated it is strongly suggested to move to the new data source as the compatibility will be removed in a future release.

--- a/website/docs/r/release.html.markdown
+++ b/website/docs/r/release.html.markdown
@@ -16,14 +16,9 @@ A Chart is a Helm package. It contains all of the resource definitions necessary
 ## Example Usage
 
 ```hcl
-data "helm_repository" "stable" {
-  name = "stable"
-  url  = "https://kubernetes-charts.storage.googleapis.com"
-}
-
 resource "helm_release" "example" {
   name       = "my-redis-release"
-  repository = data.helm_repository.stable.metadata[0].name
+  repository = "https://kubernetes-charts.storage.googleapis.com" 
   chart      = "redis"
   version    = "6.0.1"
 
@@ -65,12 +60,12 @@ The following arguments are supported:
 
 * `name` - (Required) Release name.
 * `chart` - (Required) Chart name to be installed. A path may be used.
-* `repository` - (Optional) Repository where to locate the requested chart. If is an URL the chart is installed without installing the repository.
+* `repository` - (Optional) Repository URL where to locate the requested chart.
 * `repository_key_file` - (Optional) The repositories cert key file
 * `repository_cert_file` - (Optional) The repositories cert file
 * `repository_ca_file` - (Optional) The Repositories CA File. 
 * `repository_username` - (Optional) Username for HTTP basic authentication against the repository.
-* `repository_password` - (Optional) Password for HTTP basic authentication against the reposotory.
+* `repository_password` - (Optional) Password for HTTP basic authentication against the repository.
 * `devel` - (Optional) Use chart development versions, too. Equivalent to version '>0.0.0-0'. If version is set, this is ignored.
 * `version` - (Optional) Specify the exact chart version to install. If this is not specified, the latest version is installed.
 * `namespace` - (Optional) The namespace to install the release into. Defaults to `default`

